pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -80,10 +79,11 @@ const escapeCharacter = '\\'
 // SQLTokenizer is the struct used to generate SQL
 // tokens for the parser.
 type SQLTokenizer struct {
-	rd       *strings.Reader // the "rune" reader
-	pos      int             // byte offset of lastChar
-	lastChar rune            // last read rune
-	err      error           // any error occurred while reading
+	pos      int    // byte offset of lastChar
+	lastChar rune   // last read rune
+	buf      []byte // buf holds the query that we are parsing
+	index    int    // index is the index into buf where the unread portion of the query begins.
```

Comment:
This is generally called `off` (reading offset) https://golang.org/src/bytes/buffer.go#L22. Maybe it'd be nice to use the same since we're doing something similar?

Reply:
👍